### ISSUE TITLE
FirebaseAuth.DefaultInstance.CurrentUser is not updated

### ISSUE THREAD

**Issue by QuentinGprd** [firebase/quickstart-unity#444]
Hello,

More bugs for Firebase:

## Please fill in the following fields:
Unity editor version: 2018.3.14f1
Firebase Unity SDK version: 6.2.2
Firebase plugins in use (Auth, Database, etc.): Auth, Analytics, Database, Crashlytics, Messaging
Platform you are using the Unity editor on (Mac, Windows, or Linux): MacOS Catalina 10.15 beta
Platform you are targeting (iOS, Android, and/or desktop): iOS, Android

## Please describe the issue here:
In Unity Editor, when we call `ReauthenticateAsync` successfully, we are still unable to call `UpdateEmailAsync` or update any sensitive user information after:

> This operation is sensitive and requires recent authentication. Log in again before retrying this 
request.

We need to logout and re-signin "manually".

Also, after all update operations, including `UpdateUserProfileAsync`, we need to logout and signin again to see the changes in FirebaseAuth.DefaultInstance.CurrentUser, even after restarting the Editor.

Not tested on Android or iOS.

Thanks for your support!

**Comment by DellaBitta** [comment]
Hi @QuentinGprd,

Unfortunately I wasn't able to recreate this issue.

I tested using the [auth/test_app](https://github.com/firebase/quickstart-unity/tree/master/auth/testapp) as a baseline.  In ReauthenticateAysnc I hard-coded an update to the user profile by updating the display name once the re-authentication completed.

**Line 442**, immediately after the LogTaskCompletion if {} block, add:

```return UpdateUserProfileAsync(newDisplayName: "<something>");```

This is successfully updates the profile (immediately after re-authenticating via email) without an error in my testing.

As an additional test, in UpdateUserProfileAsync I chain a task which updates the email address of the account. I then output both auth.CurrentUser and the DefaultInstance.CurrentUser when it completes.  In my tests they were both successfully updated:

**Line 303**, add:

```
auth.CurrentUser.UpdateEmailAsync("<some_test_email>")
   .ContinueWithOnMainThread(update_email_task => { 
        EnableUI();
        if (LogTaskCompletion(update_email_task, "User Email")) {
            DisplayDetailedUserInfo(auth.CurrentUser, 1);
            DisplayDetailedUserInfo(Firebase.Auth.FirebaseAuth.DefaultInstance.CurrentUser, 2);
       }
});
```

Could you update this code to make it fail in the way that you're seeing errors in your app?  Thanks!

**Comment by QuentinGprd** [comment]
I just added the following button in the auth/test_app UIHandler:

```
if (GUILayout.Button("Update DisplayName")) {
            UpdateUserProfileAsync("Kiki");
}
```
[Here is a video of the bug](https://drive.google.com/open?id=1JWJXf5q2RmxmxfHswSOH5Y6tRY83Pphj)

As you can see, after re-entering Play Mode, the previous name is still displayed. I need to logout then sign in to get the new one.

Let me know if you need anything else.

**Comment by stewartmiles** [comment]
Hi this was fixed in 6.4.0 https://firebase.google.com/support/release-notes/unity#version_640_-_august_26_2019
